config/jobs/kubernetes-sigs/azuredisk-csi-driver/azuredisk-csi-driver-config.yaml

```diff
@@ -687,6 +687,8 @@ presubmits:
               value: "true"
             - name: WINDOWS_FLAVOR
               value: "containerd-2022"
```

Comment:
after I used this config `containerd-2022`, I still got `Windows Server 2019 Datacenter` in capz e2e test, sth. wrong?
@CecileRobertMichon @marosset
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_azuredisk-csi-driver/1054/pull-kubernetes-e2e-capz-azure-disk-windows/1498155530961555456/build-log.txt

```
NAME                              STATUS   ROLES                  AGE    VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                         KERNEL-VERSION     CONTAINER-RUNTIME           LABELS
capz-jn2u-8j6rb                   Ready    <none>                 149m   v1.22.1   10.1.0.4      <none>        Windows Server 2019 Datacenter   10.0.17763.2565    containerd://1.6.0-beta.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/instance-type=Standard_D2s_v3,beta.kubernetes.io/os=windows,failure-domain.beta.kubernetes.io/region=eastus2,failure-domain.beta.kubernetes.io/zone=0,kubernetes.io/arch=amd64,kubernetes.io/hostname=capz-jn2u-8j6rb,kubernetes.io/os=windows,node.kubernetes.io/instance-type=Standard_D2s_v3,node.kubernetes.io/windows-build=10.0.17763,topology.kubernetes.io/region=eastus2,topology.kubernetes.io/zone=0
```

Reply:
seems both `AZURE_NODE_MACHINE_TYPE` and `WINDOWS_FLAVOR` do not work in CI:
https://prow.k8s.io/log?container=test&id=1498514636197072896&job=pull-kubernetes-e2e-capz-azure-disk-windows
it still creates `Windows Server 2019 Datacenter` windows node with `Standard_D2s_v3` vm size.

Reply:
For VM size the var is `NODE_MACHINE_TYPE` (without AZURE prefix)

There is no `WINDOWS_FLAVOR` in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/scripts/ci-entrypoint.sh so that explains why it's not working. I think  `CLUSTER_TEMPLATE="test/ci/cluster-template-prow-ci-version-windows-containerd-2022.yaml"` might be what you want although I don't know if you want "ci-version" (k8s build from CI). @jsturtevant is there an easier way to set the windows version?

Reply:
not with this `entrypoint.sh` script.  the default for this script is `test/ci/cluster-template-prow.yaml`, we would need to add a  new flavor like `test/ci/cluster-template-prow-ws-2022.yaml` that would use the WS2022 base images that are produced.   

There might be another way to provide the configuration at cluster generation time, either an environment or attribute?

Reply:
created an issue to track https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2139